db/transforms/operations/finish_specifying.py

```diff
@@ -80,62 +78,60 @@ def _split_missing_input_aliases_into_group_and_agg_lists(
             else:
-                alias_set_to_add_to = aliases_to_be_added_to_agg_on
-            alias_set_to_add_to.append(initial_column.alias)
+                alias_list_to_add_to = aliases_to_be_added_to_agg_on
+            alias_list_to_add_to.append(input_alias)
     else:
-        aliases_to_be_added_to_agg_on = list(
-            initial_column.alias
-            for initial_column
-            in initial_columns_not_in_summarize
-        )
+        aliases_to_be_added_to_agg_on = list(missing_input_aliases)
     return aliases_to_be_added_to_group_by, aliases_to_be_added_to_agg_on
 
 
-def _is_first_alias_generating_transform(db_query, ix_of_summarize_transform):
+def _make_sure_base_grouping_column_will_be_in_group_by_list(
+    base_grouping_alias,
+    missing_input_aliases,
+    aliases_to_be_added_to_group_by,
+):
+    """
+    We'll always want base_grouping_column in the "group-by list";
+    """
+    if base_grouping_alias in missing_input_aliases:
+        aliases_to_be_added_to_group_by.append(base_grouping_alias)
+        missing_input_aliases.remove(base_grouping_alias)
+
+
+def _get_missing_input_aliases(db_query, ix_of_summarize_transform):
     """
-    Checks if the transform is the first alias-generating transform. An alias-generating transform
-    means that it itroduces new aliases (columns) to the transform pipeline. We want to know when
-    a given alias-generating transform is the first in the pipeline, because then we can consider
-    its input aliases to be fully described by initial columns, which can be a useful
-    simplification.
+    Missing input aliases are those that are not mentioned in a summarize transform's spec. It's
+    the input aliases that we'll be automatically adding into the said spec, so that it can be
+    considered fully specified.
     """
-    prior_transforms = db_query.transformations[:ix_of_summarize_transform]
-    for prior_transform in prior_transforms:
-        alias_generating_transforms = {Summarize, SelectSubsetOfColumns}
-        is_alias_generating = \
-            type(prior_transform) in alias_generating_transforms
-        if is_alias_generating:
-            return False
-    return True
-
-
-def _get_initial_columns_not_in_summarize(db_query, summarize_transform):
-    initial_columns = db_query.initial_columns
+    summarize_transform = db_query.transformations[ix_of_summarize_transform]
     group_by_aliases = summarize_transform.grouping_input_aliases
     agg_on_aliases = summarize_transform.aggregation_input_aliases
     aliases_in_summarize = group_by_aliases + agg_on_aliases
+    input_aliases = db_query.get_input_aliases(ix_of_summarize_transform)
     return [
-        initial_column
-        for initial_column in
-        initial_columns
-        if initial_column.alias not in aliases_in_summarize
+        input_alias
+        for input_alias
+        in input_aliases
+        if input_alias not in aliases_in_summarize
     ]
```

Comment:
I like this formulation.

Reply:
Oh, diregard my comment. I thought you proposed this change.